src/test/compile-fail/E0493.rs

```diff
@@ -14,9 +14,20 @@ struct Foo {
 
 impl Drop for Foo {
     fn drop(&mut self) {}
+    //~^ NOTE destructor defined here
 }
 
-const F : Foo = Foo { a : 0 }; //~ ERROR E0493
+struct Bar {
+    a: u32
+}
+
+impl Drop for Bar {
+    fn drop(&mut self) {}
+}
+
+const F : Foo = Foo { a : 0 };
+//~^ ERROR constants are not allowed to have destructors [E0493]
+//~| NOTE constants cannot have destructors
```

Comment:
Because we're updating the span, can you move this test over to somewhere in the `src/test/ui` directory so that we can test the output?

UI tests are pretty straightforward.  The compiler runs them then compares the output with the corresponding .stderr file.  If they match, it passes.  You can look at the other .stderr files to see how it's done.